backend/src/graphql.ts

```diff
@@ -301,5 +301,52 @@ schema.mutationType({
         })
       },
     })
+    t.field('updateComment', {
+      type: 'Comment',
+      args: {
+        commentId: intArg({ required: true }),
+        body: stringArg({ required: true }),
+      },
+      resolve: async (parent, args, ctx) => {
+        const { userId } = ctx.request
+        if (!userId) throw new Error('You must be logged in to do that.')
+
+        const comment = await ctx.db.comment.update({
+          data: {
+            body: args.body,
+          },
+          where: {
+            id: args.commentId,
+          },
+        })
+
+        if (!comment) throw new Error('Comment not found.')
+        if (comment.authorId !== userId)
+          throw new Error('You do not have permission to edit this comment.')
+
+        return comment
+      },
+    })
+    t.field('deleteComment', {
+      type: 'Comment',
+      args: {
+        commentId: intArg({ required: true }),
+      },
+      resolve: async (parent, args, ctx) => {
+        const { userId } = ctx.request
+        if (!userId) throw new Error('You must be logged in to do that.')
```

Comment:
I wonder how we can translate these 🤔 I think we'll need to add the user's locale to the `context` and use it to retrieve the appropriate translation. So we may also have to move our locale files up from `/journaly/frontend/public/static/locales/**/*.json` into `/journaly`, then `/journaly/backend` could access them too. I created #106 to track and explore this

Reply:
It might still be helpful to have our backend errors in English at the point that they are thrown in things like server logs, etc. so I wonder if we could create a simple mechanism where we have a list of these errors with translations on the frontend, and we send some kind of code along with the error and then when we catch it on the frontend, map it to the correct error that has translations set up? Just thinking outloud

Reply:
It would be a bit of a conundrum for Ryan to be digging into our server logs in the middle of the night trying to fix a production fire and they're all in Punjabi, Indonesian, and Hakka Chinese 🧐 😂

Reply:
I feel like should be able to have a pretty manageable list of client-side errors that we need to display, so that's why I'm thinking something like that could perhaps work well. But open to other ideas. Great to have the discussion here but I think we could handle this in its own PR 😊( unless we come up with a solution pretty easily in the next day or so)

Reply:
For sure! I was _just_ about to comment that we could also just send an error code from the backend and have that be the key to the translation. We'd just want to make sure to maintain one source of truth for where the error message in English exists. I'll update the issue to explore this too!

Reply:
Cool sounds good!! 🙂